sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -388,8 +388,9 @@ case class AlterTableChangeColumnCommand(
     // Find the origin column from dataSchema by column name.
     val originColumn = findColumnByName(table.dataSchema, columnName, resolver)
     val validType = canEvolveType(originColumn, newColumn)
+    val collationChanged = validType && originColumn.dataType != newColumn.dataType
```

Comment:
How do you know that this is about collation change? From the code this means that only dataType is different.

Reply:
The only possible data type change is collation change or recursively changing subtype collations. Please check `canEvolveType` [function](https://github.com/apache/spark/blob/09aec04f794994655d23c83a7d9eb8a6eabe8fb8/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L476).

Reply:
It just feels wrong to write code that is specific to collations when wrapping function is called `AlterTableChangeColumnCommand` which is pretty generic name. Once we try to extend support for changing column types for anything else (e.g. set different timezone information, change decimal precision...) this `collationChanged` will no longer make sense.

Reply:
Changed to `collationChanged ` to `columnTypeChanged` to feel more generic.